main.rb

```diff
@@ -0,0 +1,128 @@
+# My module Enumerable
+module Enumerable
+  # Create #my_each, a method that is identical to #each but (obviously) does not use #each.
+  # You'll need to remember the yield statement.
+  # Make sure it returns the same thing as #each as well.
+  def my_each
+    return 'No block given' unless block_given?
+
+    for value in self do
+      yield(value)
+    end
+  end
+
+  # Create #my_each_with_index in the same way.
+  def my_each_with_index
+    return 'No block given' unless block_given?
+
+    index = 0
+    for value in self do
+      yield(value, index)
+      index += 1
+    end
+  end
+
+  # Create #my_select in the same way, though you may use #my_each in your definition (but not #each).
+  def my_select
+    return 'No block given' unless block_given?
+
+    arr = []
+    my_each do |value|
+      arr.push(value) if yield(value)
+    end
+  end
+
+  # Create #my_all? (continue as above)
+  def my_all?
+    return 'No block given' unless block_given?
+
+    result = true
+    my_each do |value|
+      return false unless yield(value)
+    end
+    result
+  end
+
+  # Create #my_any?
+  def my_any?
+    return 'No block given' unless block_given?
+
+    result = false
+    my_each do |value|
+      return true if yield(value)
+    end
+    result
+  end
+
+  # Create #my_none?
+  def my_none?
+    return 'No block given' unless block_given?
+
+    result = true
+    my_each do |value|
+      return false if yield(value)
+    end
+    result
+  end
+
+  # Create #my_count
+  def my_count
+    count = 0
+    my_each do
+      count += 1
+    end
+    count
+  end
+
+  # Create #my_map
+  def my_map_old
+    return 'No block given' unless block_given?
+
+    arr = []
+    my_each do |value|
+      arr.push(yield(value))
+    end
+    arr
+  end
+
+  # Create #my_inject
+  def my_inject(initial = 0)
+    return 'No block given' unless block_given?
+
+    total = initial
+    my_each do |x|
+      total = yield(total, x)
+    end
+    total
+  end
```

Comment:
- [ ] should have working `my_inject` **(covered by automated tests)** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject
- [ ] enumerables #my_inject gives an error when no block or argument is given
<img width="419" alt="8" src="https://user-images.githubusercontent.com/58504655/99540485-6091be80-29d1-11eb-8f8d-77bf0c6ac7be.PNG">


- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
<img width="495" alt="9" src="https://user-images.githubusercontent.com/58504655/99540567-7c956000-29d1-11eb-9954-11aa79e30092.PNG">



- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
[2, 5, 6].my_inject(:+) == [2, 5, 6].inject(:+) => 13
```
<img width="556" alt="0" src="https://user-images.githubusercontent.com/58504655/99540672-a5b5f080-29d1-11eb-85c2-e8c9c9ef0c57.PNG">